File: FIBONACCI.hs
--Fibonacci Series print
-- 1) input is int and returns int
fib :: Int -> Int
fib 0 = 0
fib 1 = 1
fib n = fib (n-1) + fib (n - 2)

--works like array to store multiple values
fibList n = map fib[0..n-1]

main =
    do
    putStrLn "Enter number of elements you want of fibonacci : "
    x <- readLn
    putStrLn "fibonacci series is :"
    print (fibList x)

File: FIBONACCI.pl
      fib(0,0).
      fib(1,1).
      fib(N,Ans):-
      N >1,
      N1 is N-1,
      N2 is N-2,
      fib(N1,A1),
      fib(N2,A2),
      Ans is A1+A2.

# fib(6,Ans).
# Ans=8.
